### PR TITLE
Improve WebSocket support

### PR DESCRIPTION
**Pull request by mentos1386**
TODO:
- [x] Update readme.md by adding warning that global interceptor only applies to http controllers, not gateways.

### CODE REVIEW

**Comment by mentos1386** [comment]
TODO: Remove Global Websockets tests, as they work as intended (limitation by NestJS that gateways don't support global interceptors)

File: test/websockets/class.gateway.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { WebSocketGateway, SubscribeMessage } from '@nestjs/websockets'
+import { RavenInterceptor } from '../../lib';
+import { UseInterceptors } from '@nestjs/common';
+
+@UseInterceptors(new RavenInterceptor())
```
Comment (mentos1386) [inline test/websockets/class.gateway.ts]: Use `{ context: 'Ws' }`

File: test/http/global.e2e.spec.ts [hunk 2]
```diff
@@ -6,7 +6,7 @@ import { GlobalModule } from './global.module';
 
 declare var global: any;
 
-describe('Global', () => {
+describe('Http:Gloval', () => {
```
Comment (mentos1386) [inline test/http/global.e2e.spec.ts]: Typo

File: lib/raven.interfaces.ts [hunk 3]
```diff
@@ -15,4 +15,5 @@ export interface IRavenInterceptorOptions {
   extra?: { [key: string]: any };
   fingerprint?: string[];
   level?: Sentry.Severity;
+  context?: 'Http' | 'Ws' | 'Rpc';
```
Comment (mentos1386) [inline lib/raven.interfaces.ts]: Document the new options

File: test/websockets/class.module.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+@Module({
+  imports: [
+    RavenModule,
+  ],
+  providers: [
+    classGateway,
+    {
+      provide: APP_INTERCEPTOR,
+      useValue: new RavenInterceptor(),
```
Comment (mentos1386) [inline test/websockets/class.module.ts]: Use `{ context: 'Ws' }`

File: lib/raven.interceptor.ts [hunk 5]
```diff
@@ -8,36 +8,68 @@ import { Observable } from 'rxjs';
...
     .pipe(
       tap(null, (exception) => {
         if (this.shouldReport(exception)) {
-          Sentry.withScope(scope => this.captureException(scope, httpRequest, userData, exception));
+          Sentry.withScope(scope => {
+            // TODO: When https://github.com/nestjs/nest/issues/1581 gets implemented switch to that
+            switch(this.options.context){
+              case 'Http': return this.captureHttpException(scope, http, exception);
+              case 'Ws': return this.captureWsException(scope, ws, exception);
+              case 'Rpc': return this.captureRpcException(scope, rpc, exception);
```
Comment (mentos1386) [inline lib/raven.interceptor.ts]: Missing tests for `Rpc`

**Comment by ricardovf** [comment]
Will this be merged?

**Pull request merged**